### DISCUSSION TITLE
Is there a prometheus endpoint?

### DISCUSSION THREAD

**Discussion by human-77** [blockscout/frontend#2376] [Q&A]
We have a private besu chain monitored with blockscout. We are usign a modified version of the [helm chart](https://github.com/blockscout/helm-charts/tree/main/charts/blockscout-stack) to run it on pods on the cloud and wanted to monitore it with prometheus. However, when [searching for it](https://github.com/blockscout/frontend/issues?q=prometheus), I could only find a merge request.

Is there a `/metrics` for the frontend? Where I can find the documentation?

Thanks in advance.

**Comment by tom2drum** [comment]
Hello. Yes, there is an endpoint to collect Prometheus metrics. It is `/node-api/metrics`, as stated in the [prometheus.yml](https://github.com/blockscout/frontend/blob/main/prometheus.yml) file.

**Reply by human-77** [reply]
Sorry for taking so long to answer.

I have seen this file, but how do I enable it? Or at least, how can I know it's actually enabled?

I tried to curl ports 9090 and 3000 and got nothing. What I'm not seeing?

**Reply by tom2drum** [reply]
Did you pass `PROMETHEUS_METRICS_ENABLED=true` variable?

**Reply by human-77** [reply]
I added this variable to my enviroment variables and still got the bellow result:

```bash
$ kubectl exec -it my-pod-97aa96fbcf-ktxfm -- bash
root@my-pod-97aa96fbcf-ktxfm:/# curl localhost:9090/node-api/metrics
^C
root@my-pod-97aa96fbcf-ktxfm:/# curl localhost:9090/node-api/metrics
curl: (7) Failed to connect to localhost port 9090 after 0 ms: Couldn't connect to server
root@my-pod-97aa96fbcf-ktxfm:/# curl localhost:3000/node-api/metrics
curl: (7) Failed to connect to localhost port 3000 after 0 ms: Couldn't connect to server
root@my-pod-97aa96fbcf-ktxfm:/# curl localhost:443/node-api/metrics
root@my-pod-97aa96fbcf-ktxfm:/# curl my-pod-svc:443/node-api/metrics

root@my-pod-97aa96fbcf-ktxfm:/# curl my-pod-svc:3000/node-api/metrics

root@my-pod-97aa96fbcf-ktxfm:/# curl my-pod-svc:9090/node-api/metrics
^C
root@my-pod-97aa96fbcf-ktxfm:/# curl my-pod-svc/node-api/metrics
```

**Reply by tom2drum** [reply]
And in what port your frontend app is running?

**Reply by human-77** [reply]
Once again, sorry again for the delay.

It's running on the ports that I tried to curl above.

Basically we are using a modified image of the helm chart ( a few extra fields for our cloud). I added the variable that you gave me to the env of the chart, and yet, nothing.

The funny thing is: when I try to curl port 3000, I get the return "200", which shows up as a empty line:

```bash
root@my-pod-97aa96fbcf-ktxfm:/# curl my-pod-svc:3000/node-api/metrics

```

When I tried to look for anything related to prometheus in the official repository for the frontend, the only thing I found was a [merge request](https://github.com/blockscout/frontend/pull/1781) that seemed to only make minor changes, including adding the 200 response, rather than the [automatic 404](https://github.com/blockscout/frontend/pull/1781/commits/92a9a5e676657dc0717b0b52504fd455ae2e992c).

This made me believe that maybe the `/metrics` was still in development.

What I'm not seeing.

**Reply by tom2drum** [reply]
> The funny thing is: when I try to curl port 3000, I get the return "200", which shows up as a empty line:

So, maybe there is no metric there yet?

But that's odd since once enabled, Prometheus should collect default Node.js metrics - https://github.com/blockscout/frontend/blob/db405c1c13b9ca5e75126bfec74707b7e9a866c3/pages/api/metrics.ts#L6

Can you run your custom frontend image locally and test this endpoint?

**Reply by human-77** [reply]
We run it using:

```bash
$ docker run -p 3000:3000 --env-file .env ghcr.io/blockscout/frontend:latest
```

And then call it using the provided address. It worked.

We have no idea why it's not working on the pod, but it may be a chart thing.

Thank you for your help

**Comment by human-77** [comment]
Sorry for taking so long to answer.

I have seen this file, but how do I enable it? Or at least, how can I know it's actually enabled?

I tried to curl ports 9090 and 3000 and got nothing. What I'm not seeing?
